### ISSUE TITLE
"CUDNN_STATUS_BAD_PARAM" raised when running MNIST expert tutorial on GPU

### ISSUE THREAD

**Issue by amineHorseman** [tensorflow/tensorflow#2107]
I tried the code from MNIST expert tutorial and it works only on cpu. When I try to execute it in the gpu I get a session crash.
The code for begginers tutorial however works well in both the gpu and cpu.

### Environment info
- OS: Ubuntu 16.04
- CUDA: 7.5
- cuDNN: 5
- Pip: 8.1.1
- Python: 2.7.11
- Anaconda:  4.0.0 (64-bit)
- Tensorflow: 0.8.0 (running on conda environment)
- GPU: GeForce GTX 860M

### Steps to reproduce
1. conda create -n tensorflow python=2.7
2. pip install --ignore-installed --upgrade https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.8.0-cp27-none-linux_x86_64.whl

### What have you tried?
1. I tried the MNIST tutorial for begginer and it works well on both cpu and gpu
2. I tried the MNIST tutorial for expert on cpu and it works well
3. I tried the MNIST tutorial for expert on gpu and its not working, here's the output:

```
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcurand.so locally
Extracting MNIST_data/train-images-idx3-ubyte.gz
Extracting MNIST_data/train-labels-idx1-ubyte.gz
Extracting MNIST_data/t10k-images-idx3-ubyte.gz
Extracting MNIST_data/t10k-labels-idx1-ubyte.gz
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:900] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_init.cc:102] Found device 0 with properties: 
name: GeForce GTX 860M
major: 5 minor: 0 memoryClockRate (GHz) 1.0195
pciBusID 0000:01:00.0
Total memory: 2.00GiB
Free memory: 1.67GiB
I tensorflow/core/common_runtime/gpu/gpu_init.cc:126] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_init.cc:136] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:755] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 860M, pci bus id: 0000:01:00.0)
F tensorflow/stream_executor/cuda/cuda_dnn.cc:427] could not set cudnn filter descriptor: CUDNN_STATUS_BAD_PARAM
Aborted (core dumped)
```

## Additional information (the output of `ls -l /path/to/cuda/lib/libcud*`):
```
-rw-r--r-- 1 root root  189170 Apr 22 15:05 libcudadevrt.a
lrwxrwxrwx 1 root root      16 Apr 22 15:05 libcudart.so -> libcudart.so.7.5
lrwxrwxrwx 1 root root      19 Apr 22 15:05 libcudart.so.7.5 -> libcudart.so.7.5.18
-rwxr-xr-x 1 root root  311596 Apr 22 15:05 libcudart.so.7.5.18
-rw-r--r-- 1 root root  558020 Apr 22 15:05 libcudart_static.a
lrwxrwxrwx 1 root root      17 Apr 22 15:05 libcuinj32.so -> libcuinj32.so.7.5
lrwxrwxrwx 1 root root      20 Apr 22 15:05 libcuinj32.so.7.5 -> libcuinj32.so.7.5.18
-rwxr-xr-x 1 root root 5396088 Apr 22 15:05 libcuinj32.so.7.5.18
lrwxrwxrwx 1 root root       13 Apr 22 16:38 libcudnn.so -> libcudnn.so.5
lrwxrwxrwx 1 root root       17 Apr 22 16:38 libcudnn.so.5 -> libcudnn.so.5.0.4
-rwxrwxr-x 1 root root 59823168 Mar 22 10:37 libcudnn.so.5.0.4
```

**Comment by vrv** [comment]
This is because you have cudnn v5, which requires installing from source.

See: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#requirements for info

**Comment by philipperemy** [comment]
If you have the error `"CUDNN_STATUS_BAD_PARAM"`, make sure that this problem does not come from your code.

For example if you give an empty array as an input.

**Comment by ducta-qc** [comment]
I also encounter this error. In my case, the forward pass works normally but the backward pass I filtered unsuitable samples (convert from [`origin_batch_size`, h, w, 3] -> [`keep_samples`, h, w, 3]) for training and keep them backprop through graph, it makes on some batches has zero sample when run the backward pass (keep_samples=0). I solved my problem by settings contributed loss to zero when `keep_samples` is zero.
